cassandra/cqltypes.py

```diff
@@ -832,9 +832,12 @@ def serialize_safe(cls, items, protocol_version):
         buf.write(pack(len(items)))
         inner_proto = max(3, protocol_version)
         for item in items:
-            itembytes = subtype.to_binary(item, inner_proto)
-            buf.write(pack(len(itembytes)))
-            buf.write(itembytes)
+            if item is None:
+                buf.write(int32_pack(-1))
```

Comment:
I think this (and also two more times below) should be pack instead of int32_pack.
I don't think we actually support of test protocol version < 3, but if we use int16_pack in pack(len(keybytes)), I don't see any reason why packing -1 should be different.

I don't think this matters in practice because nobody uses cql version < 3, but let's make this clean.

Reply:
I have no idea if this is correct for version 2 of the protocol, but it can be tested (on an antique version of Cassandra). What bothered me was the inconsistency - one branch of the if() uses "pack" and the other "int32_pack", and that must be wrong (I don't know which branch is more wrong...).

Reply:
Ok, the current code is the correct one - I'll fix my change. Sections 6.10-6.13 of aforementioned document describe that each item of collection is serialized as `[short bytes]`, not `[bytes]`.